src/Operations.jl

```diff
@@ -777,7 +777,7 @@ function with_dependencies_loadable_at_toplevel(f, mainctx::Context, pkg::Packag
         write_env(localctx, display_diff = false)
         will_resolve && build_versions(localctx, new)
         sep = Sys.iswindows() ? ';' : ':'
-        withenv(f, "JULIA_LOAD_PATH" => "$tmpdir$sep$(Types.stdlib_dir())")
+        withenv(f, "JULIA_LOAD_PATH" => "@|$tmpdir$sep$(Types.stdlib_dir())")
```

Comment:
This is certainly the right fix to add the current env to the path (i.e. look in parent directories for something that looks like a project or a git repo). I'm not quite sure that I understand the scenario that @staticfloat is hitting or why this is the right fix. Can you explain a bit?

Reply:
Yes; this is for the `BinaryProvider` tests; one of the things I wanted to test was that the whole `Pkg.*` machinery works, so I have a package called `LibFoo.jl` [embedded within `BinaryProvider/test`](https://github.com/JuliaPackaging/BinaryProvider.jl/tree/master/test/LibFoo.jl).  With `Pkg2`; getting `Pkg.build()` and `Pkg.test()` to be able to reach this package is pretty easy; [you can just set `JULIA_LOAD_PATH` and be on your way](https://github.com/JuliaPackaging/BinaryProvider.jl/blob/74f50756a0bd96e874fa82d36933cf867af0abe1/test/runtests.jl#L771-L782). 0.7 makes it a little more complicated because doing so will allow `LibFoo` to be loaded, but `LibFoo` will then not be able to load `Compat` or `BinaryProvider`.  What I ended up doing was adding a `Project.toml` for `LibFoo` as well, and that convinced `Pkg3` to expose `Compat` and `BinaryProvider` to `LibFoo.jl` (Which is such an amazingly simple and staightforward fix, I love it) but that only worked when running directly from the command line, e.g. `julia test/runtests.jl` from within the `BinaryProvider` directory would be able to pass all tests, including those of the nested `LibFoo.jl`.  Running `Pkg.test("BinaryProvider")` would fail and be unable to load `LibFoo.jl`.

Reply:
Ah, I think I want `$sep` here.